To: vim_dev@googlegroups.com
Subject: Patch 7.3.230
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.230
Problem:    ":wundo" and ":rundo" don't unescape their argument.  (Aaron
            Thoma)
Solution:   Use FILE1 instead of XFILE.
Files:      src/ex_cmds.h


*** ../mercurial/vim73/src/ex_cmds.h	2010-08-20 11:09:02.000000000 +0200
--- src/ex_cmds.h	2011-06-20 04:44:13.000000000 +0200
***************
*** 788,794 ****
  EX(CMD_rubyfile,	"rubyfile",	ex_rubyfile,
  			RANGE|FILE1|NEEDARG|CMDWIN),
  EX(CMD_rundo,		"rundo",	ex_rundo,
! 			NEEDARG|EXTRA|XFILE),
  EX(CMD_rviminfo,	"rviminfo",	ex_viminfo,
  			BANG|FILE1|TRLBAR|CMDWIN),
  EX(CMD_substitute,	"substitute",	do_sub,
--- 788,794 ----
  EX(CMD_rubyfile,	"rubyfile",	ex_rubyfile,
  			RANGE|FILE1|NEEDARG|CMDWIN),
  EX(CMD_rundo,		"rundo",	ex_rundo,
! 			NEEDARG|FILE1),
  EX(CMD_rviminfo,	"rviminfo",	ex_viminfo,
  			BANG|FILE1|TRLBAR|CMDWIN),
  EX(CMD_substitute,	"substitute",	do_sub,
***************
*** 1078,1084 ****
  EX(CMD_wsverb,		"wsverb",	ex_wsverb,
  			EXTRA|NOTADR|NEEDARG),
  EX(CMD_wundo,		"wundo",	ex_wundo,
! 			BANG|NEEDARG|EXTRA|XFILE),
  EX(CMD_wviminfo,	"wviminfo",	ex_viminfo,
  			BANG|FILE1|TRLBAR|CMDWIN),
  EX(CMD_xit,		"xit",		ex_exit,
--- 1078,1084 ----
  EX(CMD_wsverb,		"wsverb",	ex_wsverb,
  			EXTRA|NOTADR|NEEDARG),
  EX(CMD_wundo,		"wundo",	ex_wundo,
! 			BANG|NEEDARG|FILE1),
  EX(CMD_wviminfo,	"wviminfo",	ex_viminfo,
  			BANG|FILE1|TRLBAR|CMDWIN),
  EX(CMD_xit,		"xit",		ex_exit,
*** ../vim-7.3.229/src/version.c	2011-06-20 00:53:10.000000000 +0200
--- src/version.c	2011-06-20 05:02:40.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     230,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
201. When somebody asks you where you are, you tell them in which chat room.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
